ctl/shared_ptr.h

```diff
@@ -0,0 +1,462 @@
+// -*-mode:c++;indent-tabs-mode:nil;c-basic-offset:4;tab-width:8;coding:utf-8-*-
+// vi: set et ft=cpp ts=4 sts=4 sw=4 fenc=utf-8 :vi
+#ifndef COSMOPOLITAN_CTL_SHARED_PTR_H_
+#define COSMOPOLITAN_CTL_SHARED_PTR_H_
+#include "enable_if.h"
+#include "is_void.h"
+#include "new.h"
+#include "unique_ptr.h"
+
+// TODO(mrdomino): move
+#ifndef __cplusplus
+#define _CTL_ATOMIC(x) _Atomic(x)
```

Comment:
So `_Atomic` is really a C11 thing. The C++ compiler doesn't support it. It's only able to be supported in C++23 because I think it uses a macro to turn it into `std::atomic<size_t>` which naturally requires schlepping in a million libcxx headers. Should we just use the GCC compiler builtins in this header to make things simpler? Also why check `__cplusplus` when it's a C++ only header?

Reply:
Basically just cribbed this stanza from the 5 other places that pattern is used (`rg '_Atomic\(x\)'`.) Indeed, no reason to check `__cplusplus` here.

The STL atomic thing is indeed a nightmare. Maybe we'll wind up writing our own stdatomic.h, with blackjack, etc.

Meanwhile, in the new PR I just use the GCC builtins as you suggest.